src/gen.rs

```diff
@@ -80,8 +80,16 @@ fn empty_generics() -> ast::Generics {
 
 fn rust_id(ctx: &mut GenCtx, name: &str) -> (String, bool) {
     let token = parse::token::Ident(ctx.ext_cx.ident_of(name));
-    if token.is_any_keyword() || "bool" == name {
+    if token.is_any_keyword() ||
+        name.contains("@") ||
+        name.contains("?") ||
+        name.contains("$") ||
+        "bool" == name
+    {
         let mut s = name.to_owned();
+        s = s.replace("@", "_");
+        s = s.replace("?", "_");
+        s = s.replace("$", "_");
         s.push_str("_");
```

Comment:
Potentially we just want to append the underscore if it's a rust keyword, right? This should also work though.